### ISSUE TITLE
Accessing strong order memory space with vector instructions

### ISSUE THREAD

**Issue by ZeyueShen** [riscv/riscv-isa-manual#1501]
"Is it possible for vector instructions not to support accessing memory regions that have the 'strong order' address space attribute? If vector instructions do not support this, how should software respond when it attempts to access memory with such an attribute?"

**Comment by gfavor** [comment]
Yes.  One way to approach this is to have PMA I/O regions not support vector memory accesses. Violations of a PMA region's supported attributes would, as always, result in an Access Fault on the offending instruction.

Note that you also have to worry about a vector instruction and its memory accesses straddling I/O and non-I/O regions - whether at the level of all of the instruction's accesses, or each individual memory access, or each individual element.  Whatever your choices are, if you're taking an Access Fault in the middle of a vector instruction, then the standard precise exception and "vstart" semantics apply.

**Comment by aswaterman** [comment]
Don't forget, element accesses can be reordered for all vector instructions except for the indexed-ordered ones.  So, while the instructions are all usable on strongly ordered regions, some of them might not do what you want.
